src/lib/geospatial.ts

```diff
@@ -0,0 +1,43 @@
+interface LatLng {
+  lat: number
+  lng: number
+}
+
+const EARTH_RADIUS_IN_METERS = 6371e3 // https://en.wikipedia.org/wiki/Earth_radius#Mean_radius
+
+/**
+ * Convert angular distance from degrees to radians
+ *
+ * @param {number} degrees - decimal degrees
+ * @returns {number} radians
+ */
+const toRadians = (degrees: number): number => degrees * (Math.PI / 180)
+
+/**
+ * Calculates the haversine (spherical) distance between two geographic points.
+ * Arguments are supplied as `LatLng` objects, with coordinates specified
+ * as `lat` and `lng` properties in decimal degrees.
+ *
+ * See:
+ * https://en.wikipedia.org/wiki/Haversine_formula
+ * https://www.movable-type.co.uk/scripts/latlong.html
+ *
+ * @param {LatLng} point1 - an object with `lat` and `lng` properties
+ * @param {LatLng} point2 - an object with `lat` and `lng` properties
+ * @returns {number} Distance between point1 and point2, in meters
+ */
+const haversineDistance = (point1: LatLng, point2: LatLng): number => {
+  const φ1 = toRadians(point1.lat)
+  const φ2 = toRadians(point2.lat)
+  const Δφ = toRadians(point2.lat - point1.lat)
+  const Δλ = toRadians(point2.lng - point1.lng)
```

Comment:
Yep, [they're legal](https://mothereff.in/js-variables#Δφ) in ES6

Reply:
Symbol support is cool :) but its also a little domain specific (for those reading at a glance) vs plain words. (Also hard to figure out how to type from a standard keyboard.)

Reply:
I'm ok with a bit of an artful touch 🎨, especially in a pace that will likely only be worked on by people that know this domain.

Reply:
Ooo didn't know about ctrl+command+space!